### ISSUE TITLE
Deserialization of MySQL BOOLEAN type does not work

### ISSUE THREAD

**Issue by HubertLegec** [mirromutth/r2dbc-mysql#119]
I've database column defined like below (it's MySQL 5.7):
`purging_enabled BOOLEAN NOT NULL`
When I try to read it as Boolean type:
`row.get("purging_enabled", Boolean.class)`
I get exception:
```
Cannot decode value of type class java.lang.Boolean for type 1 with collation 63
		at dev.miku.r2dbc.mysql.codec.DefaultCodecs.decode(DefaultCodecs.java:132)
		at dev.miku.r2dbc.mysql.MySqlRow.get0(MySqlRow.java:95)
		at dev.miku.r2dbc.mysql.MySqlRow.get(MySqlRow.java:63)
```

**Comment by HubertLegec** [comment]
Similar situation occurs when try to store bool value in column of type `BOOLEAN`.

**Comment by mirromutth** [comment]
Hello, thanks for your report.

It is the bug of boolean codec, for now it supports only `BIT(1)` as `Boolean` (it accecpts only 0 and 1).
The `BOOLEAN` is actually `TINYINT(1)` and it accepts values between 0 and 9 (0 is false, others are true).

It will be supported today.

**Comment by mirromutth** [comment]
It is supported by PR and published to `0.8.2.BUILD-SNAPSHOT`. Target version is `0.8.2.RELEASE`.

> If want to use snapshot versions, add Sonatype snapshots repository to Maven file, see also README
